interpreter/syntax/types.ml

```diff
@@ -16,7 +16,7 @@ type extern_type =
   | ExternMemoryType of memory_type
   | ExternGlobalType of global_type
 
-type pack_size = Pack8 | Pack16 | Pack32
+type pack_size = Pack8 | Pack16 | Pack32 | Pack64 | Pack8x8 | Pack16x4 | Pack32x2
```

Comment:
Ah, hm, so the pack_size is supposed to merely encode the size of something and has been reused as that in other contexts such as the new `extend` instructions (see main repo). So I think we need to keep SIMD shapes out of it.

Also, in the reference types proposal, the existing `Load` and `Store` instructions are already restricted to deal with plain numeric types only, since in the presence of reference types they can no longer accept all value types.

To be compatible with these parallel developments, I'd suggest introducing separate `LoadSimd` and `StoreSimd` instructions that carry the following ops:
```
type pack_simd = Pack8x8 | Pack16x4 | Pack32x2 | PackSplat of pack_size
type simd_loadop = (pack_simd * extension) memop
type simd_storeop = [] memop
```
(`[]` is the empty type, such that the memop's `sz` field can only be None for stores.)

Reply:
Got it, in the latest commit I added a new `SimdLoad` ast node, and a `simd_loadop`. Tweaked the type definition slightly from your suggestion here.
I don't think a separate store is required, there is only 1 way of storing a v128, and it can be already be handled correctly.

Reply:
I think a separate Store would be preferable, a) for symmetry, and b) because the pack_size present with numeric stores doesn't make sense for it.

Reply:
(Also, once we merge reference types, the regular Store will only work for num_types anyway.)

Reply:
Sg, made the change.
I tweak the `simd_loadop` type a little bit as well - since packsize is the size of something loaded from memory, it made the use of it consistent with `loadop`, each `simd_loadop` is a a tuple of `(pack_size * pack_simd)`, the fst encodes the size to be loaded, the snd is a simd specific shape to describe what to do after loading (splat/extend).

I couldn't do `[] memop` as you suggested, it was a syntax error. I searched around and found https://ocaml.org/releases/4.10/htmlman/emptyvariants.html so I created a `type empty = |`. Lmk if I should change it.

Reply:
Ah, I wasn't aware that polymorphic variants can't be empty. Odd. But since we don't actually care it's a variant it's sufficient to define
```
type empty
```
with no RHS.

Reply:
Good piont, done in #345.